### ISSUE TITLE
Problem with multi threading with Pool.map

### ISSUE THREAD

**Issue by BobBDE** [coleifer/peewee#2811]
Hello,

Thank you for the library, it is quite nice to use :) 

I have a strange problem when I run select in multiple thread in the same time. I have a select that is executed in parallel in multithreading that works. But if I add a select (or a create) before the creation of the multi process, it doesn't work anymore. 

Here is a simple code to show what is happening : 
```
import multiprocessing
from peewee import CharField, Model, MySQLDatabase

class TestTable(Model):
    id = CharField(primary_key=True, max_length=36)
    text = CharField()

    class Meta:
        database = MySQLDatabase('test', user='test', password='test',host='test', port=3306)

def init_table():
    TestTable.drop_table()
    TestTable.create_table()
    for i in range(0, 1000):
        TestTable.create(id=str(i), text=f'text_{i}')
    print('Table created')

def run():
    # if we comment the select outside the sub processes and the create table was not done before it works
    list(TestTable.select())
    run_in_processes()

    # If we call the method run_in_processes in a thread, it works
    # thread = threading.Thread(target=run_in_processes)
    # thread.start()
    # thread.join()

def run_in_processes():
    i = 0
    # run select in sub process (4 in parallel)
    with multiprocessing.Pool(processes=4) as pool:
        for _ in pool.map(simple_select, list(range(0, 16))):
            print(f'Count success : {i}')
            i += 1

    print('Success')

def simple_select(_):
    list(TestTable.select())

init_table()
run()
```

This code will crashed with the error `peewee.InternalError: Packet sequence number wrong - got X expected 1'`.
But if I run the same code without the table creation (table already created) and without the select in the `run` method, the selects in executed in the sub process will correctly work. 

Also if in the `run` method we call the `run_in_processes` using a new thread it works even if request on the table were done before. 

I tried with the `PooledMySQLDatabase` or the `thread_safe` parameter but it doesn't have a impact on the problem. 
I tried this is peewee V 3.15 and 3.17.

Basically if a request was made before the requests in the sub thread, there is a problem. 
Do anyone have an idea ?

**Comment by coleifer** [comment]
When you use multiprocessing you're likely forking a new process, which means it's copying the file descriptors for the sockets used to communicate with the server. The protocol may be stateful and in general it's a terrible idea to open a connection and then fork and use that connection in multiple child processes.

What you should do is ensure:

* You do not open a connection before forking, or
* You close your connection before forking, or
* You close the connection in the child processes and then reopen it.

**Comment by BobBDE** [comment]
Thank you for those information, it is clearer now and the closing connection before opening the new process works. 
Thanks again !
